lib/wlib/strings/DynamicString.cpp

```diff
@@ -0,0 +1,482 @@
+/**
+ * @file DynamicString.cpp
+ * @brief DynamicString is a class that provides dynamic strings and functions for dynamic strings
+ *
+ * @author Bob Wei
+ * @date November 25, 2017
+ * @bug No known bugs
+ */
+
+#include "DynamicString.h"
+
+namespace wlp {
+
+    /**
+     * Default constructor creates string with no characters
+     */
+    DynamicString::DynamicString() {
+        m_buffer = malloc<char>();
+        m_buffer[0] = '\0';
+        len = 0;
+    }
+
+    /**
+     * Constructor creates string using character array
+     *
+     * @param str char string
+     */
+    DynamicString::DynamicString(const char *str) {
+        m_buffer = nullptr;
+        auto size = (size_type) strlen(str);
+        m_buffer = malloc<char[]>(static_cast<size_type>(size + 1u));
+        strncpy(m_buffer, str, size);

```

Comment:
Since `m_buffer` will always have enough space for `str`, just a regular `strcpy` is more performant than `strncpy`. You'll have to double check whether `strcpy` also copies the null character because I forget